Add toFloat() and remove() function

itsme6582 opened the pull request:
Add toFloat() as a String method to the documentation

itsme6582 committed:
Update firmware.md

Add String.toFloat()

kennethlimcp commented:
Hmm.. interesting. I wonder why has this not been in the docs for such a long time since the implementation.

itsme6582 committed:
Add String.remove() and reword toFloat()

After adding toFloat() I looked at the String class documentation for Arduino and found remove() is missing from the Particle documentation as well.

Add String.remove()
Change toFloat() instructions to match Arduino documentation

itsme6582 commented:
It's easy to miss small things like this when dealing with big projects,
especially with looming deadlines.

On Thu, Aug 27, 2015 at 1:44 AM, Kenneth Lim notifications@github.com
wrote:

> Hmm.. interesting. I wonder why has this not been in the docs for such a
> long time since the implementation.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/spark/docs/pull/144#issuecomment-135300799.

## 

Jesse Weiland
jesse.weiland@gmail.com

kennethlimcp commented:
Looks good. I will try out and see if the instructions are accurate. @m-mcgowan might be faster for you to read through the additions?

m-mcgowan commented:
:+1:  done, get's my vote.   I can add some API tests in firmware also.

kennethlimcp commented:
@itsme6582 looks like this is good to merge but I would need the 2 commits squashed into one. Are you familiar with that? Otherwise i can do it :)

itsme6582 commented:
Sure. I'll get to it sometime over Labor Day weekend.

@itsme6582 https://github.com/itsme6582 looks like this is good to merge
but I would need the 2 commits squashed into one. Are you familiar with
that? Otherwise i can do it :)

—
Reply to this email directly or view it on GitHub
https://github.com/spark/docs/pull/144#issuecomment-136778570.

kennethlimcp commented on src/content/reference/firmware.md:
```diff
@@ -4301,6 +4319,21 @@ Parameters:
 
+### toFloat()
+
+Converts a valid String to a float. The input string should start with a digit. If the string contains non-digit characters, the function will stop performing the conversion. For example, the strings "123.45", "123", and "123fish" are converted to 123.45, 123.00, and 123.00 respectively. Note that "123.456" is approximated with 123.46. Note too that floats have only 6-7 decimal digits of precisio ...
```
@itsme6582 can we update to : 

Converts a valid String to a float. The input string should start with a digit. If the string contains non-digit characters, the function will stop performing the conversion. For example, the strings "123.45", "123", and "123fish" are converted to 123.45, 123.00, and 123.00 respectively. Note that "123.456" is approximated with 123.46. Note that a float has only 6-7 decimal digits of precision and anything behind might be truncated.

kennethlimcp commented:
@itsme6582 sorry for the delay. Thanks once again! :smile_cat:
